### ISSUE TITLE
Training Hypterparameters of PixelSnail for VQ-VAE experiments

### ISSUE THREAD

**Issue by fostiropoulos** [rosinality/vq-vae-2-pytorch#49]
I am using 4x Nvidia V100 and I am not able to get a batch size larger than 32 for the hyperparameters of this paper for training on the top codes. I have also changed the loss to discretized mixtures of logistics similar to the actual PixelCNN++ and PixelSnail implementation. The authors mention a batch size of 1024 which seems unreal to reach. Does this implementation of PixelSnail use more layers than the one reported in the VQVAE2 paper? 

I am not able to make the mapping between this implementation and the one described in the appendix of VQVAE 2 to correctly configure it to replicate their results. Any help appreciated. 

![image](https://user-images.githubusercontent.com/4337024/95422896-004d2e80-08f5-11eb-9ff8-a9518c979b6a.png)

**Comment by rosinality** [comment]
Actually the network used in the paper is much larger than the default model in this implementation.

**Comment by fostiropoulos** [comment]
Yes I would initially have thought so. I can only think of being able to train such a large model on a TPU. Do you have any insights on how it could have been done?

**Comment by rosinality** [comment]
Maybe they have used tpus or large amount of gpus. Anyway replicating the model training in the paper will be very hard (actually practically impossible) with a few number of gpus.
